### PR TITLE
Edit user redo

### PR DESCRIPTION
**Pull request by EricBelongea**
## Description

Please include a description of what was implemented. What was changed? What was added or removed if anything? 

This is the redone user#edit/update branch. The other was having many merge conflicts so I created a new branch for ease, Blake did a lot of the lift here, so much credit goes to him.

## Type of change
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Describe testing implemented and what all it covers. 

- [x]  fully tests happy and sad paths

## Checklist

- [x] code has been self reviewed
- [x] code runs without any errors
- [x] all tests pass

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -19,6 +19,25 @@ def edit
...
+        flash[:error] = response[:error].join(', ')
+        redirect_back(fallback_location: edit_user_path(current_user))
+      end
+    else
+      session[:user_id] = params[:id]
+      flash[:error] = "Passwords Must Match"
+      redirect_back(fallback_location: edit_user_path(current_user))
+    end
+  end
+
```
Comment (ttakahashi1591) [inline app/controllers/users_controller.rb:40]: This looks great and the sessions will be useful for when we start building Oauth. We may need to add something for password confirmation on the BE and this is something we can look into.

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -3,7 +3,9 @@
...
-<br><%= button_to "Add Pet", add_dog_path %>
-<br><%= button_to "View Pets", dogs_path %>
-<br><%= button_to "Discover Socials", user_socials_path %>
\ No newline at end of file
+<div id="dashboard_nav" >
+  <p><%= button_to "Edit Profile", edit_user_path(@user.user_id), method: :get %></p>
+  <p><%= button_to "Add Pet", add_dog_path %></p>
+  <p><%= button_to "View Pets", dogs_path %></p>
+  <p><%= button_to "Discover Socials", socials_path, method: :get %></p>
+</div>
```
Comment (ttakahashi1591) [inline app/views/users/show.html.erb:11]: This navigation setup looks great!

File: spec/features/users/edit_spec.rb [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+      expect(page).to have_content("Editing James Sullivan's Profile")
+      expect(page).to have_field :name, with: 'James Sullivan'
+      expect(page).to have_field :email, with: 'sully@gmail.com'
+      expect(page).to have_field :password 
+      expect(page).to have_field :password_confirmation  
+      expect(page).to have_button('Save')  
+      expect(page).to have_link('Back to User Dashboard')  
+
+      # No use of updating a profile photo as I thought we were still deciding on whether or not to use images
```
Comment (ttakahashi1591) [inline spec/features/users/edit_spec.rb:16]: I agree here!

**Review by ttakahashi1591 — COMMENTED** [review]
This all looks great and I really like the thorough testing with sad paths that you included.

**Review by delaneymiranda1 — APPROVED** [review]
Everything looks great. Nice thorough testing.

**Pull request merged**